### PR TITLE
Remove use of py2to3.h

### PR DESCRIPTION
**Pull request by aaronayres35**
Another step towards dropping python 2 support.  This PR removes the use of `py2to3.h` in `_interpolate.cpp`.

### CODE REVIEW

File: scimath/interpolate/py2to3.h [hunk 1]
```diff
@@ -1,256 +0,0 @@
...
-          static struct PyModuleDef moduledef = { \
-            PyModuleDef_HEAD_INIT, name, doc, -1, methods, }; \
-          ob = PyModule_Create(&moduledef);
-#else
-  #define Py2to3_MOD_ERROR_VAL
-  #define Py2to3_MOD_SUCCESS_VAL(val)
-  #define Py2to3_MOD_INIT(name) void init##name(void)
-  #define Py2to3_MOD_DEF(ob, name, doc, methods) \
-          ob = Py_InitModule3(name, methods, doc);
-#endif
```
Comment (rahulporuri) [inline scimath/interpolate/py2to3.h:256]: FTR, this is the only relevant bits of code in `py2to3.h` which are used in `_interpolate.cpp`

File: scimath/interpolate/_interpolate.cpp [hunk 2]
```diff
@@ -260,21 +259,18 @@ static PyMethodDef interpolate_methods[] = {
...
 {
     PyObject* m;
-    Py2to3_MOD_DEF(
-        m,
-        "_interpolate",
-        "A few interpolation routines.\n",
-        interpolate_methods
-    );
+    static struct PyModuleDef moduledef = { \
+        PyModuleDef_HEAD_INIT, "_interpolate", "A few interpolation routines.\n", -1, interpolate_methods, }; \
```
Comment (rahulporuri) [inline scimath/interpolate/_interpolate.cpp:266]:
```suggestion
    static struct PyModuleDef moduledef = {
        PyModuleDef_HEAD_INIT,
        "_interpolate",
        "A few interpolation routines.\n",
        -1,
        interpolate_methods,
    };
```

**Review by rahulporuri — APPROVED** [review]
LGTM with one minor formatting suggestion. It might be good to wait for another pair of eyes on this PR @aaronayres35

**Comment by rahulporuri** [comment]
@mdickinson Can you make time to review this PR? Not at all urgent.

**Review by mdickinson — APPROVED** [review]
LGTM. I don't see any related compiler warnings. (I do see one _unrelated_ compiler warning, but that's off-topic for this PR.)

**Pull request merged**